Model/reviews.post.model.js

```diff
@@ -0,0 +1,47 @@
+const db = require("../db/connection.js");
+
+exports.selectReviewPost = (review_id, patchInfo) => {
+  const toPatchThisKey = Object.keys(patchInfo);
+  const validReviewKeys = [
+    "title",
+    "review_body",
+    "designer",
+    "review_img_url",
+    "votes",
+    "category",
+    "owner",
+  ];
+  return db
+    .query(`SELECT * FROM reviews WHERE review_id = ${review_id.reviews_id}`)
+    .then((reviewsIdData) => {
+      if (reviewsIdData.rows.length === 0) {
+        return Promise.reject({ status: 404, msg: "Not Found" });
+      } else {
+        let currentValue = reviewsIdData.rows[0][toPatchThisKey[0]];
+        let queryStr = "UPDATE reviews";
+        let newValue = 0;
+        const createdQueryObj = {};
+        if (
+          typeof patchInfo[toPatchThisKey[0]] === "number" &&
+          toPatchThisKey[0] === "votes"
+        ) {
+          newValue = currentValue += patchInfo[toPatchThisKey[0]];
+        } else {
+          return Promise.reject({ status: 400, msg: "Bad Request" });
+        }
+        if (validReviewKeys.includes(toPatchThisKey[0])) {
+          queryStr += ` SET ${toPatchThisKey[0]} = $1`;
```

Comment:
I really hate how this syntax on line 33 opens you up to injection - can you make this use params also

Reply:
Sure not a problem